Stats Widget: Fix dashboard styles impacted by widget styling

dognose24 opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to #75360

## Proposed Changes

* Extract modified layout styles only for the Odyssey Stats page to an independent file.
* Use minimal base styles built from Calypso to reduce the impact on base elements, e.g., `<p>` and `<ul>`.

|Before|After|
|-|-|
|<img width="1916" alt="截圖 2023-04-18 下午9 42 05" src="https://user-images.githubusercontent.com/6869813/232798230-bb3cc997-f809-4054-88a1-8e6a33b4ec4e.png">|<img width="1927" alt="截圖 2023-04-18 下午9 41 12" src="https://user-images.githubusercontent.com/6869813/232798302-e5589939-542f-46e3-b629-1368155ea8cf.png">|

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Open your local Jetpack site.
* Check out the branch of the Jetpack to the latest `trunk`.
* Build Jetpack if necessary: `jetpack build plugins/jetpack`.
* Run `STATS_PACKAGE_PATH=/path/to/jetpack/projects/packages/stats-admin yarn dev` under `calypso/apps/odyssey-stats`.
* Navigate to `/wp-admin/index.php?odyssey_widget=1`.
* Ensure the `Dashboard` page styling works the same as the page without the feature flag `odyssey_widget=1`.

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

dognose24 committed:
Extract layout styles only for Stats page

dognose24 committed:
Use base styles to replace complete Calypso styles

kangzj reviewed (approved):
Thanks for fixing this. It works well for me! 👍

jsnmoon reviewed (approved):
Works well! Let's use this PR as an opportunity for a design review 🙂

jsnmoon reviewed (commented):
Since we're expanding the scope of this PR to include designer reviews, I wanted to point out a two things:

<img width="494" alt="image" src="https://user-images.githubusercontent.com/4044428/232909076-404b2466-245d-42b8-9e32-3ceb7a982359.png">

1. The current top posts/pages & referrers are not linked. Could we make these individual top posts and referrers deeply link into Odyssey Stats?

<img width="504" alt="image" src="https://user-images.githubusercontent.com/4044428/232909273-1d789828-0de0-4dce-9eeb-e3f90951e0d0.png">

2. The text in the Akismet module overflows to two lines. This likely needs a design solution.

kangzj commented:
Another thing is we'll need to format the numbers. I don't see any code handling that.

![image](https://user-images.githubusercontent.com/1425433/232910374-c4919a97-e500-409f-a7dc-9ca1c804ef02.png)

![image](https://user-images.githubusercontent.com/1425433/232910508-f91702ae-f86c-4f5b-9040-b271e9c71807.png)

kangzj commented:
I'm happy that we address these feedback in separate PRs.

Hey @filipanoscampos feel free to comment here and we'll follow up. Thanks 🙇

keoshi reviewed (approved):
LGTM!

ederrengifo commented:
@jsnmoon @kangzj regarding the design issues:

1. Is this related to what we were exploring in the P2? To make these links point to their details instead of the posts?
2. Is there a way to force both cards to go to two lines simultaneously? Maybe some breakpoint?
<img width="511" alt="Screen Shot 2023-04-19 at 09 48 46" src="https://user-images.githubusercontent.com/10525007/233113534-fac5fa9c-2875-4042-94df-f99310e55aa9.png">
3. I agree we should limit the posts and referrers to only 5
4. Do you need anything from the design regarding the numbers formating? Or is this a development issue to solve?

ederrengifo reviewed (commented):
LGTM!

dognose24 committed:
Limit highlight items to 5

dognose24 committed:
Format view count numbers

jsnmoon commented:
> 1. Is this related to what we were exploring in the P2? To make these links point to their details instead of the posts?

Yes! That's exactly what I was thinking.

> 2. Is there a way to force both cards to go to two lines simultaneously? Maybe some breakpoint?

This is possible, but I'll let @dognose24 make the final call for this.

> 3. I agree we should limit the posts and referrers to only 5

👍 

> 4. Do you need anything from the design regarding the numbers formating? Or is this a development issue to solve?

I don't think so! Definitely a development issue.

-----

@dognose24: Feel free to address these issues in this PR or in a follow-up PR.

dognose24 committed:
Make item linkable and update hover styling

dognose24 committed:
Merge branch 'trunk' into update/stats_widget_impacted_styles

dognose24 commented:
@kangzj @jsnmoon @keoshi @ederrengifo Thanks for all your reviews and feedback! Since this PR was expanded to use the final design review, I would prefer to fix all significant issues here.

> The current top posts/pages & referrers are not linked. Could we make these individual top posts and referrers deeply link into Odyssey Stats?

Updated in [31b00a8](https://github.com/Automattic/wp-calypso/pull/75897/commits/31b00a8460bd09bb841fb2dff8cc27a28316b99b).

> it shows at most 5 items for posts and referrers.

Fixed in [bf34c12](https://github.com/Automattic/wp-calypso/pull/75897/commits/bf34c126ad62839c9eec983267217fc64a513e4f).

> Another thing is we'll need to format the numbers.

Fixed in [4e43898](https://github.com/Automattic/wp-calypso/pull/75897/commits/4e438982c156254210e175d75135568bdfb27110), and the format for the module card numbers [is there already](https://github.com/Automattic/wp-calypso/blob/update/stats_widget_impacted_styles/apps/odyssey-stats/src/widget/modules.jsx#L15).

> Is there a way to force both cards to go to two lines simultaneously? Maybe some breakpoint?

Since the whole dashboard layout would change on different breakpoints, the window breakpoint may not absolutely relate to the Stats Widget frame width. Let's tackle this issue in a follow-up PR!

ederrengifo commented:
@jsnmoon could we hold this change until we develop stage 1 of the widget upgrade?

jsnmoon commented on apps/odyssey-stats/package.json:
```diff
@@ -29,6 +29,7 @@
 		"@automattic/calypso-polyfills": "workspace:^",
 		"@automattic/calypso-url": "workspace:^",
 		"@automattic/components": "workspace:^",
+		"@wordpress/icons": "workspace:^",
```
How did you add this dependency? It looks like the yarn lockfile wasn't updated and is causing build failures. I recommend using `yarn add some-dependency/here` to avoid this.

jsnmoon commented on apps/odyssey-stats/package.json:
```diff
@@ -29,6 +29,7 @@
 		"@automattic/calypso-polyfills": "workspace:^",
```
I think we'll need to add `@automattic/calypso-color-schemes` as a dependency.

ederrengifo reviewed (commented):
I think we need to adjust the line-height of the items list and be sure we follow these specs:
<img width="807" alt="Screen Shot 2023-04-19 at 15 16 26" src="https://user-images.githubusercontent.com/10525007/233190014-7c4f7b55-3693-48b1-a0a0-c323ef18a7bb.png">

dognose24 committed:
Fix item padding

jsnmoon commented on apps/odyssey-stats/src/widget/highlights.jsx:
```diff
@@ -6,7 +8,57 @@ import useTopPostsQuery from '../hooks/use-top-posts-query';
+		<a
+			href={ postAndPageLink( odysseyStatsBaseUrl, siteId, item.id ) }
+			rel="noopener noreferrer"
+			title={ translate( 'View highlight post or page', {
+				textOnly: true,
+				context: 'View highlight post or page',
+			} ) }
+			aria-label={ translate( 'View highlight post or page', {
+				textOnly: true,
+				context: 'View highlight post or page',
+			} ) }
```
```suggestion
			title={ translate( 'View detailed stats for %(title)s', {
				args: {
					title: item.title
				},
				textOnly: true,
				comment: 'Text for anchor linking to a stats page for a given post/page',
			} ) }
```

No need to add a separate `aria-label` if a `title` is already provided.

jsnmoon reviewed (commented):
<img width="537" alt="image" src="https://user-images.githubusercontent.com/4044428/233189697-1f065ba9-d046-4ec4-94f6-e8fe072077c4.png">

Could we update the styling of the `No activity this period` notice to more closely resemble our design specifications? c6yabERlJE06tKLECfpnch-fi-6580-138157&t=kProk6bdXfHgnwac-4

jsnmoon commented:
> @jsnmoon could we hold this change until we develop stage 1 of the widget upgrade? (@ederrengifo)

Could you clarify which change you're referring to? Was it the linked highlights?

dognose24 committed:
Fix view all link margin top

dognose24 committed:
Fix dependencies

dognose24 committed:
Fix item link title

dognose24 committed:
Update empty state with styling

dognose24 commented:
> Could we update the styling of the No activity this period notice to more closely resemble our design specifications?

@jsnmoon Nice catch! The empty state is fresh now.

<img width="597" alt="截圖 2023-04-20 上午5 28 17" src="https://user-images.githubusercontent.com/6869813/233204201-3d63e846-f4a4-46fd-951f-cf980ae80112.png">

kangzj committed:
changed chart axis labels smaller

do not wrap

kangzj commented:
I added a commit 145536a to make the font size smaller for axis labels, which fix some wrapping issue.

kangzj reviewed (approved):
LGTM

kangzj committed:
yarn dedupe

kangzj committed:
merge from trunk

kangzj reviewed (approved):
I can see all feedback are addressed. I'll go ahead and merge this, but if you have any more feedback, feel free to drop them here and we'll fix in follow up PRs. @jsnmoon @ederrengifo @keoshi

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7935173

Thank you @dognose24 for including a screenshot in the description! This is really helpful for our translators.
